src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -381,27 +385,26 @@ private bool AgeCookies(string domain)
                 // Each domain will be cut accordingly.
                 remainingFraction = (float)m_maxCookies / (float)m_count;
             }
-            lock (m_domainTable)
+            lock (m_domainTable.SyncRoot)
             {
-                foreach (KeyValuePair<string, PathList> entry in m_domainTable)
+                foreach (DictionaryEntry entry in m_domainTable)
```

Comment:
This is going to allocate a boxed object for each entry in the dictionary (see #7539, #9050, #10247). It'd be better to do something like:

```c#
// Manual use of IDictionaryEnumerator instead of foreach to avoid DictionaryEntry box allocations.
IDictionaryEnumerator e = m_domainTable.GetEnumerator();
while (e.MoveNext())
{
    if (domain == null)
    {
        tempDomain = (string)e.Key;
        pathList = (PathList)e.Value;
    }
    ...
}
```